### PR TITLE
ROX-12434: do not use stackrox.io flavor for tagged builds

### PR DESCRIPTION
**Pull request by johannes94**
## Description

Nightly CI operator tests are failing because it tries to pull images from stackrox.io which is no longer available. This PR changes it to use "development_build" image flavor instead for the test.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Review by porridge — APPROVED** [review]
Screenshot from 2022-09-14 17-20-10
:ship: !!
Looks good from operator PoV, I haven't looked at the failures to see whether they are related.

**Comment by vikin91** [comment]
Yes, they are related, but it should be easy to address them - the error messages suggest what is the issue.

File: tests/roxctl/bats-tests/helpers.bash [hunk 1]
```diff
@@ -226,10 +227,13 @@ image_reference_regex() {
   local component="$2"
   local version="${3:-$any_version}"
 
-  case $registry_slug in
-    quay.io)
+  case "$registry_slug" in
+    quay.io/rhacs-eng)
```
Comment (vikin91) [inline tests/roxctl/bats-tests/helpers.bash:231]: I think, there may be still some test cases that would use the old slug `quay.io`. We need to either change the assertions to assert against `quay.io/rhacs-eng` or  change this line to handle both cases: 
```suggestion
    quay.io|quay.io/rhacs-eng)
```

**Comment by msugakov** [comment]
I want to open this for early review while waiting for the CI results.

**Comment by janisz** [comment]
Can you split it into series of smaller PRs?

File: tests/roxctl/bats-tests/local/central-generate-development-postgres.bats [hunk 2]
```diff
@@ -5,10 +5,11 @@ load "../helpers.bash"
 out_dir=""
 
 setup_file() {
+  # remove binaries from the previous runs
+  [[ -n "$NO_BATS_ROXCTL_REBUILD" ]] || rm -f "${tmp_roxctl}"/roxctl*
```
Comment (vikin91) [inline tests/roxctl/bats-tests/local/central-generate-development-postgres.bats:9]: Cool for speeding-up local testing!

File: tests/roxctl/bats-tests/local/central-generate-release.bats [hunk 3]
```diff
@@ -88,7 +93,7 @@ teardown() {
 }
 
 @test "roxctl-release central generate k8s should not support --central-db-image" {
-  run roxctl-development central generate k8s pvc --output-dir "$out_dir" --central-db-image example.com/central-db:1.2.5
+  run roxctl-release central generate k8s pvc --output-dir "$out_dir" --central-db-image example.com/central-db:1.2.5
```
Comment (vikin91) [inline tests/roxctl/bats-tests/local/central-generate-release.bats:96]: Good catch!

**Review by vikin91 — APPROVED** [review]
Quite many changes in the bats tests, but I skimmed through them and they look good!
The main change is about hiding the dev flavor in release builds instead of preventing to use it, right?

**Comment by msugakov** [comment]
Sorry, @janisz saw your commit just now. I try making semantic commits. It is similar to stacked PRs but all the context is in one PR. Try reviewing per commit.

**Comment by janisz** [comment]
@msugakov got it but there's no way to approve by commit.

**Comment by msugakov** [comment]
@janisz Yes, it's only to be sold entirely.

**Pull request merged**